From c5a6ecc6b8986988e1d2129d575e9ccad23f0689 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 10 Jul 2024 14:18:52 +0300
Subject: [PATCH] [FSW-2704] fs1: use flag for force recovery

Instead of going into recovery, bootrom first
needs to check if force recovery is set. If
recovery is set then based on validation pass
bit, it needs to jump to proper recovery image.

Change-Id: I65b9bf6bd610876936f649c3fb3daed6a119b860
---

diff --git a/app/boot_rom.c b/app/boot_rom.c
index 02eb3ef..b7cc6fe 100644
--- a/app/boot_rom.c
+++ b/app/boot_rom.c
@@ -15,6 +15,7 @@
 	.config_fuse = 0,
 	.config2_fuse = 0,
 	.recovery_flag = 0,
+	.force_recovery = 0,
 	.validation_pass = 0,
 	.valid_manifest_idx = 0,
 	.fw_upd_switch_manifest = 0,
@@ -49,7 +50,7 @@
 
 	boot_rom_wait_for_signal();
 
-	boot_rom_check_n_enter_recovery();
+	boot_rom_check_recovery();
 
 	boot_rom_init_boot_info();
 
diff --git a/common/common.c b/common/common.c
index c09aef2..e1e1192 100644
--- a/common/common.c
+++ b/common/common.c
@@ -121,7 +121,7 @@
 	/* Empty Body */
 }
 
-void boot_rom_check_n_enter_recovery(void)
+void boot_rom_check_recovery(void)
 {
 	bool override = true;
 
@@ -130,6 +130,6 @@
 
 	if (((!boot_rom_boot_desc_init_done(HL_FW_SPI_MANIFEST_0)) &&
 		(!boot_rom_boot_desc_init_done(HL_FW_SPI_MANIFEST_1))) || (!override))
-		execute_recovery();
+		boot_rom_global.force_recovery = true;
 }
 
diff --git a/include/boot_rom.h b/include/boot_rom.h
index 914db5a..fd288c0 100644
--- a/include/boot_rom.h
+++ b/include/boot_rom.h
@@ -88,6 +88,7 @@
 	uint32_t config2_fuse;
 	uint8_t recovery_flag;
 	uint8_t validation_pass;
+	uint8_t force_recovery;
 	uint8_t valid_manifest_idx;
 	uint8_t fw_upd_switch_manifest;
 	uint8_t efuse_read;
@@ -123,7 +124,7 @@
 void boot_rom_fake_fuse(void);
 void boot_rom_check_rma_enabled(void);
 void boot_rom_wait_for_signal(void);
-void boot_rom_check_n_enter_recovery(void);
+void boot_rom_check_recovery(void);
 void boot_rom_bringup_ppboot(void);
 void boot_rom_init_boot_info(void);
 void boot_rom_tpm_spi_global_ctx_init(struct spi_ctx *tpm_spi);
